core/src/test/scala/caliban/execution/ExecutionSpec.scala

```diff
@@ -203,6 +203,35 @@ object ExecutionSpec extends DefaultRunnableSpec {
           api.interpreter.flatMap(_.execute(query, None, Map("term" -> StringValue("search")))).map(_.asJson.noSpaces)
         )(equalTo("""{"data":{"getId":null}}"""))
       },
+      testM("field function") {
+        import io.circe.syntax._
+
+        case class Character(name: String = "Bob")
+        case class Test(character: Field => Character)
+        val api = graphQL(RootResolver(Test(field => {
+          Character()
+        })))
+        val query = """query test { character { name } }"""
+
+        assertM(
+          api.interpreter.flatMap(_.execute(query, None, Map())).map(_.asJson.noSpaces)
+        )(equalTo("""{"data":{"character":{"name":"Bob"}}}"""))
+      },
+      testM("field function with input") {
+        import io.circe.syntax._
+
+        case class NameInput(name: String)
+        case class Character(name: String)
+        case class Test(character: Field => (NameInput => Character))
```

Comment:
I think the parenthesis are not needed, can you confirm? `Field => NameInput => Character`